go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -376,6 +377,69 @@ type managementPortEntry struct {
 	config *managementPortConfig
 }
 
+// getEnvNameFromResourceName gets the device plugin env variable from the device plugin resource name.
+func getEnvNameFromResourceName(resource string) string {
+	res1 := strings.ReplaceAll(resource, ".", "_")
+	res2 := strings.ReplaceAll(res1, "/", "_")
+	return "PCIDEVICE_" + strings.ToUpper(res2)
+}
+
+// getDeviceIdsFromEnv gets the list of device IDs from the device plugin env variable.
+func getDeviceIdsFromEnv(envName string) ([]string, error) {
+	envVar := os.Getenv(envName)
+	if len(envVar) == 0 {
+		return nil, fmt.Errorf("unexpected empty env variable: %s", envName)
+	}
+	deviceIds := strings.Split(envVar, ",")
+	return deviceIds, nil
+}
+
+// handleDevicePluginResources tries to retrieve any device plugin resources passed in via arguments and device plugin env variables.
+func handleDevicePluginResources() error {
+	mgmtPortEnvName := getEnvNameFromResourceName(config.OvnKubeNode.MgmtPortDPResourceName)
+	deviceIds, err := getDeviceIdsFromEnv(mgmtPortEnvName)
+	if err != nil {
+		return err
+	}
+	config.OvnKubeNode.DPResourceDeviceIdsMap = make(map[string][]string)
+	config.OvnKubeNode.DPResourceDeviceIdsMap[config.OvnKubeNode.MgmtPortDPResourceName] = deviceIds
+	klog.V(5).Infof("Setting DPResourceDeviceIdsMap for %s using env %s with device IDs %v", config.OvnKubeNode.MgmtPortDPResourceName, mgmtPortEnvName, deviceIds)
+	return nil
+}
+
+// takeDevicePluginInterface takes a device ID from a list of device IDs
+func takeDeviceId(deviceIds []string) (string, []string) {
+	ifsLen := len(deviceIds)
+	ret := (deviceIds)[ifsLen-1]
+	(deviceIds) = (deviceIds)[:ifsLen-1]
+	return ret, deviceIds
+}
+
+// handleDevicePluginInterfaces tries to retrieve any device plugin interfaces to be used by the system such as the management port.
+func handleDevicePluginInterfaces() error {
+	if config.OvnKubeNode.MgmtPortNetdev == "" {
+		var deviceId string
+		if len(config.OvnKubeNode.DPResourceDeviceIdsMap[config.OvnKubeNode.MgmtPortDPResourceName]) > 0 {
+			deviceId, config.OvnKubeNode.DPResourceDeviceIdsMap[config.OvnKubeNode.MgmtPortDPResourceName] = takeDeviceId(config.OvnKubeNode.DPResourceDeviceIdsMap[config.OvnKubeNode.MgmtPortDPResourceName])
+		} else {
+			return fmt.Errorf("insufficient device IDs for resource: %s", config.OvnKubeNode.MgmtPortDPResourceName)
+		}
+		ifName, err := util.GetSriovnetOps().GetNetDevicesFromPci(deviceId)
```

Comment:
You should handle SF as well. Also, should we take into account VF VDPA?
Check https://github.com/ovn-org/ovn-kubernetes/pull/3222/files#diff-7664be4fc9831cf07ad96809560316efd267b946033659d4e0973b8ceaa674cf `getNetdevName` function. I think You can rewrite original `getVFNetdevName` so it can be used for both pod request and resource request.

Reply:
I would need your PR on SF merged in first.
1) For the SF support (sriovnet changes and such)
2) There will be merge conflicts if I were to change cni.go::getVFNetdevName()
How do you want to coordinate this?